### ISSUE TITLE
Trying to use the lib with TanStack/Table

### ISSUE THREAD

**Issue by giraud** [DevExpress/testcafe-react-selectors#199]
We have been using testcafe and testcafe-react-selectors for a long time now in our end-to-end tests.
It is a great lib, thank you for that.

We recently tried to test a new page that is using a tanstack Table with the cells containing our custom Checkbox component.
Unfortunately some of our tests are failing randomly. It happens in normal mode or in native-automation mode. 
It is not always failing in the same location, so we know the selectors are correct. And occasionally(rarely) the test succeed.

our selectors:
![image](https://github.com/DevExpress/testcafe-react-selectors/assets/239612/bba9308d-7d69-4dbb-9e02-32ea20df1d51)

The UserRoleChecbox is problematic, the findReact is not working as expected.

I don’t know how we get into these situations, and my only repro case is our application.
So I tried to debug the lib with lot of logs.
What I found is that sometimes the child is correct and have the expected hierarchy and sometimes not.

For example, here my child is totally empty (the expanded FiberNode is component.child, it should have other children):
![log3](https://github.com/DevExpress/testcafe-react-selectors/assets/239612/d99f323d-f9fe-408f-93b6-18da2d59c2f8)

in that case I found that the alternate version can get me the correct child.

after digging, I found that the problem happens when the child component has a flag equals to 8. From what I’ve seen, it should correspond to `ContentReset`. And if I use the alternate child in that case, my tests are passing.

I changed the code to this:
![image](https://github.com/DevExpress/testcafe-react-selectors/assets/239612/8dc7ded4-20f5-40f4-bb70-751709217960)

I don’t know if it is the correct approach or if I am doing something wrong in my tests.
what do you think ?

versions: testcafe@2.6.2 & testcafe-react-selectors@5.0.3

**Comment by giraud** [comment]
maybe the same then #146 ?

**Comment by aleks-pro** [comment]
Hello @giraud ,

Thank you for your kind words. We appreciate them.

To find the cause of the issue, we need to reproduce it on our side. Could you please share a simple project sample with test code using which we can reproduce the issue?

**Comment by giraud** [comment]
yes, I understand, but that is my problem right now: having a small reproducible app :(.
